Automattic/jetpack
# Add 'View all purchases' and 'Activate a product' buttons in My Plan card footer.

#### Changes proposed in this Pull Request:

As part of the License-based Checkout project, this PR adds 2 new buttons in the My Plan page. (See screenshot below).

- The "Activate a product" button points to the Jetpack user license activation flow (`/wp-admin/admin.php?page=jetpack#/license/activation`).
- The "View all purchases" link is an external link that leads users to `https://wordpress.com/me/purchases`.

#### Implementation notes:
- None of the buttons will be displayed if Jetpack is not active or we are in offline mode or if this is an Atomic site.
- The "View all purchases" link will only show if the site has a paid plan or product, OR if the user has an unactivated license key(s).
- The "Activate a product" button will show only if the user has an unactivated license key(s), Otherwise the "Recommendations" button will show (Which was there prior to this PR).

![image (3)](https://user-images.githubusercontent.com/11078128/142660657-081f3ed2-408c-468a-9722-615a698654c2.png)

#### Jetpack product discussion


Project Thread: p1HpG7-cSg-p2
Figma Design: ceKxiVna5wfpmwFBsojrAx-fi-2219%3A24680

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






- Spin up a site using this Jetpack branch. 
- Connect Jetpack 
- Go to `/wp-admin/?page=jetpack#/dashboard#/my-plan` (aka "My plan" page)

- If your user account **does not** have any available license keys and your site **does not** have any paid plan or product(s), you should **only** see the "Recommendations" button (on the right).
![Screenshot on 2021-11-19 at 13-41-33](https://user-images.githubusercontent.com/11078128/142674825-d9834c82-b7f6-49e8-b838-7a88918f5790.png)

- If your user account **does not** have any available license keys and your site **does** have a paid plan or product(s), you should see the "View all purchases" link (on the left) and the "Recommendations" button (on the right).
![Screenshot on 2021-11-19 at 13-46-25](https://user-images.githubusercontent.com/11078128/142675524-bf2ed9ef-f450-4bf3-8e03-f007a70ac56b.png)

- If your user account **does**  have one or more available(detached) license keys, then you should see the "View all purchases" link (on the left) and the "Activate a product" button (on the right).
![Screenshot on 2021-11-19 at 13-50-57](https://user-images.githubusercontent.com/11078128/142675875-ee7d3dab-bcfa-4705-94ba-171a49aa3750.png)

Using [Redux DevTools](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd) extension in your browser, you can manually add or remove the number of `detached` licenses by dispatching the following object: (adjust the `detached:` value)
```
{
	type: 'JETPACK_LICENSING_USER_LICENSE_COUNTS_UPDATE',
	counts: {
		attached: 0,
		detached: 0,
		revoked: 0,
		not_revoked: 0,
	}
}
```

Then you can also add or remove your site's paid plan or product from within Store Admin.

**Using Redux DevTools and Store Admin you can test the various scenarios described above.**

Verify the "View all purchases" link and the "Activate a product" button each link to the correct location.

Verify the buttons look and behave reasonably on mobile and desktop viewport sizes.
